### ISSUE TITLE
Feature Request: Map a @ModelAttribute to a "param"

### ISSUE THREAD

**Issue by slaout** [springfox/springfox#2653]
I have a lot of APIs starting with /api/parents/{parentId}

* /api/parents/1/childs
* /api/parents/1/childs/42
* /api/parents/1/entities
* /api/parents/1/entities/a

Each of these APIs must get the parent from database, given its ID, return 404 if not found, check authorizations, etc.

To make the work more generic and transparent, I'm using the solution provided here:  
https://stackoverflow.com/questions/24758969/spring-mvc-common-param-in-all-requests/52100415#52100415  
Basically, a @ControllerAdvice will catch the (@PathVariable long parentId) parameter, do the heavy work, and then place the database entity as a @ModelAttribute.  
Then, all APIs just have to declare a (@ModelAttribute ParentDTO parent) parameter, "and voilà !".

Now, comes the swagger documentation generation...

By default, it expands ALL the members of ParentDTO as query parameters.

I added @ApiModelProperty(hidden = true) to all members of ParentDTO, but set @ApiModelProperty(name = "parentId") on the ID field.

Now, the Swagger-UI shows only the code field of the project, with a name matching the actual parameter in the URL... But insist on defining "parentId" as a "query" argument, and not a "path" parameter.  
Because it does not know that the @ModelAttribute is in fact a disguised @PathParam.  
And we of course can't put both annotations on the parameter: Spring would not allow that.

I found no way to override this behavior.

Here is the result in Swagger-UI:

![image](https://user-images.githubusercontent.com/6317877/44916024-ed8ac780-ad34-11e8-8c19-d93e7637f03a.png)

Could you offer a way to override the in="query" to be in="path"?

Or is there a more straightforward way to do so?

**Comment by dilipkrish** [comment]
There is a simpler way to do this (without exercising the code), Leaving everything the same, you might try this instead

```java
class ParentDTO {
    //Annotate it with ApiParam instead
    @ApiParam(value = "Enter an id", required = true, type = "path")
    private String parentId;
    //Continue hiding other fields
    @ApiParam(value = "Some hidden field", required = true, hidden=true)
    private String someHiddenField;
...
```

**Comment by slaout** [comment]
Nope: I tryed and it's not working.  
The "type" is for the data type ("integer", etc.).

**Comment by dilipkrish** [comment]
@slaout yes sorry for that ... I shouldve tested the code. Yes ApiParam doesnt support path parameters unfortunately. Let me get back on this

**Comment by dilipkrish** [comment]
@rehevkor5 you're issue is most likely because you have `enableUrlTemplating(true)` in your docketing

**Comment by rehevkor5** [comment]
@dilipkrish I don't.

**Comment by rehevkor5** [comment]
@dilipkrish Sure. Here is a cleaned-up version of the code:

Controller
```
    @ApiOperation(value = "description here")
    @RequestMapping(value = "/{userId}/attributes/{attributeNames}", method = RequestMethod.GET,
            produces = "application/json")
    public ResponseModel getAttributes(
            @Valid Param param) {
        return service.handleRequest(param);
    }
```

Param class:
```
package com.mystuff;

import io.swagger.annotations.ApiParam;
import org.apache.commons.lang.builder.ToStringBuilder;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.List;

public class Param {
    @ApiParam(value = "description",
            required = true, allowMultiple = true, allowableValues = "dynamic[namespace.attribute_metadata]")
    @NotNull
    @Valid
    private List<@AttributeMatcher String> attributeNames;

    @ApiParam(value = "description", required = true)
    @Valid
    @IsValidUserId
    private String userId;

    public List<String> getAttributeNames() {
        return attributeNames;
    }

    public void setAttributeNames(List<String> attributeNames) {
        this.attributeNames = attributeNames;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }
}
```
